numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -384,6 +384,7 @@ PyUFunc_NegativeTypeResolver(PyUFuncObject *ufunc,
 
     /* The type resolver would have upcast already */
     if (out_dtypes[0]->type_num == NPY_BOOL) {
+        /* 2013-12-05, 1.9 */
         if (DEPRECATE("numpy boolean negative (the unary `-` operator) is "
```

Comment:
Not your job, but just noticed that there is a copy paste error here, The message should ask to use `~` I think.